pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -415,6 +454,60 @@ def queue_view(self):
         """
         return self.view_queues()
 
+    @property
+    def executor(self) -> Union[Executor, None]:
+        """
+        Executor to execute the job object this server object is attached to in the background.
+
+        Returns:
+            concurrent.futures.Executor:
+        """
+        if not self.run_mode.executor and self._executor is not None:
+            self._executor = None
+        return self._executor
+
+    @executor.setter
+    def executor(self, exe: Union[Executor, None]):
+        """
+        Executor to execute the job object this server object is attached to in the background.
+
+        Args:
+            exe (concurrent.futures.Executor):
+        """
+        if isinstance(exe, Executor):
+            self.run_mode.executor = True
+        elif exe is None:
+            self.run_mode.modal = True
```

Comment:
```suggestion
        elif exe is None:
            if self.executor is not None:
                self.run_mode.modal = True
            # else pass, it's already None and you're setting it to None
```

I know I wrote this, but I expressed some concern then and I am still pretty sure it's bad. E.g. consider that a user thinks "oh shoot, I forgot to disable the executor" and tries to do it manually after some other run-mode-modifier:

```python
job.interactive_open()
job.server.executor = None
```

Now they've interactive opened but the run mode is modal!

I _think_ this suggestion now avoids that by first going through the `executor` property, which will force the value over to `None` if the run mode has changed, but I'll need to write up some tests to be sure it's doing what I imagine.

Reply:
It's also probably that the whole setter can now be refactored to have a more readable logic flow, but that's hard for me to see in review+suggestion mode

Reply:
I added an alternative suggestion above - only switch the `run_mode` to `modal` when the `run_mode` is still set to `executor` otherwise the `run_mode` is not modified.

Reply:
> I added an alternative suggestion above 

I don't see this, but that sounds fine too. It _should_ be formally equivalent, as any time `server.executor is not None` I expect `server.run_mode.executor`. Anything that stops a None-executor being set to more None from updating the run mode will satisfy me.

Reply:
Ah, I see it. It was just "below" for me and not "above". Hard to read with all the layered "suggestion" bits, but what you describe here sounds fine and I trust that's what the suggestions result in.